### PR TITLE
Add addNestedFields() to Form

### PR DESCRIPTION
**Pull request by trowski**
Adds a method to `Form` to insert an array of values into the form, preserving array keys.

### CODE REVIEW

File: src/Form.php [hunk 1]
```diff
@@ -49,6 +49,18 @@ public function addField(string $name, string $content, ?string $contentType = n
...
 
+    /**
+     * Adds each member of the array as an entry for the given key name. Array keys are persevered.
+     *
+     * @param array<string> $fields
+     */
+    public function addFields(string $name, array $fields): void
+    {
+        foreach ($fields as $key => $content) {
+            $this->addField(\sprintf('%s[%s]', $name, $key), $content);
```
Comment (kelunik) [inline src/Form.php]: This is PHP specific syntax, no?

File: src/Form.php [hunk 2]
```diff
@@ -49,6 +49,18 @@ public function addField(string $name, string $content, ?string $contentType = n
         $this->fields[] = new FormField($name, BufferedContent::fromString($content, $contentType));
     }
 
+    /**
+     * Adds each member of the array as an entry for the given key name. Array keys are persevered.
+     *
+     * @param array<string> $fields
+     */
+    public function addFields(string $name, array $fields): void
```
Comment (kelunik) [inline src/Form.php]: Based on our usual naming, I'd expect this to add a repeated field, not nested fields based on the array keys.

File: src/Form.php (hunk 1 above)
Comment (trowski) [reply src/Form.php]: No, it seems this is a de-facto "standard" for arrays in query strings. I found several libraries supporting it. Here's one for example in python: https://github.com/bernii/querystring-parser

File: src/Form.php (hunk 2 above)
Comment (trowski) [reply src/Form.php]: I agree. `addArrayField`? Or something else?

Comment (kelunik) [reply src/Form.php]: Maybe `addNestedFields`?

Comment (trowski) [reply src/Form.php]: Works for me. Updated.

File: src/Form.php (hunk 1 above)
Comment (kelunik) [reply src/Form.php]: Should it support any depth then?

Comment (trowski) [reply src/Form.php]: Sure, probably uncommon, but that makes sense. Updated.

**Review by kelunik — APPROVED** [review]

**Pull request merged**